main.js

```diff
@@ -137,17 +132,159 @@ function addToCartButton(product, itemList) {
   addToCartButton.textContent = 'Add to Cart'
   product.appendChild(addToCartButton)
   addToCartButton.addEventListener('click', function(event){
-    addToCart(product, shoppingCart, itemList)
+    addToCart(product, shoppingCart, items)
     updateCartButton(shoppingCart)
+    createCart(getCartItems(shoppingCart, items))
+  })
+}
+
+function updateCartButton(cartSize) {
+  var count = 0
+  for (var i = 0; i < cartSize.length; i++) {
+    if (cartSize[i] != undefined) {
+      count += cartSize[i].quantity
+    }
+  }
+  document.getElementById('count').textContent = count
+}
+
+function goToCartButton() {
+  var $cartButton = document.getElementById('cart-icon')
+  $cartButton.addEventListener('click', function(event) {
+    var $id = document.getElementById('cart').id
+    var id = getView(views, $id)
+    swapToView(views[id], views)
   })
 }
 
+goToCartButton()
+
+function createCart(cartList) {
+  var $shoppingCart = document.getElementById('shopping-cart')
+  var $totalPrice = document.getElementById('total-price')
+  $shoppingCart.innerHTML=''
+  for (var i = 0; i < cartList.length; i++) {
+    var currentProduct = cartList[i]
+    var $product = renderItem(currentProduct)
+    $product.appendChild(addSelector(shoppingCart, $product))
+    $shoppingCart.appendChild($product)
+  }
+  totalPrice(shoppingCart)
+  createGenericButton($totalPrice, buttons, 'checkoutButton')
+  createGenericButton($shoppingCart, buttons, 'returnButton')
+}
+
+function totalPrice(cartList) {
+  var total = 0
+  for (var i = 0; i < cartList.length; i++) {
+    if(cartList[i] != undefined) {
+      total += (cartList[i].price * cartList[i].quantity)
+    }
+  }
+  document.getElementById('total-price').textContent = 'Total: $' + total.toFixed(2)
```

Comment:
this function should return a total, and then where you call it, that's where you update the DOM.